### PR TITLE
Added ability to toggle mute

### PR DESCRIPTION
**Pull request by amckee**
I've updated the documentation for this new option as well.

### CODE REVIEW

**Comment by amckee** [comment]
I forgot a debugging line before creating this. Will try again.

File: pa_volume.c [hunk 1]
```diff
@@ -92,10 +94,20 @@ static void read_callback(pa_context *context,
         if(new_info.volume.channels == 0) {
           new_info.volume.channels = 2;
         }
-        pa_volume_t channel_volume =
-          CLAMP_VOLUME((pa_volume_t)(volume*PA_VOLUME_NORM));
-        pa_cvolume_set(&new_info.volume, new_info.volume.channels,
-                       channel_volume);
+        if(toggle_mute){
+          if( pa_cvolume_is_muted(&new_info.volume)){
+            fprintf(stderr, "will unmute\n");
```
Comment (rhaas80) [inline pa_volume.c:99]: the fprintf() should not be there.

File: pa_volume.c [hunk 2]
```diff
@@ -92,10 +94,20 @@ static void read_callback(pa_context *context,
...
+            pa_cvolume_reset(&new_info.volume, new_info.volume.channels);
+          } else {
+            fprintf(stderr, "will mute\n");
```
Comment (rhaas80) [inline pa_volume.c:102]: no leftover debug code please.

File: pa_volume.c [hunk 3]
```diff
@@ -311,8 +324,11 @@ static void parse_args(int argc, char **argv)
     client = argv[optind++];
   if(optind < argc) {
     char *endptr = NULL;
+    if(strcmp(argv[optind], "toggle") == 0){
+      toggle_mute = true;
+    }
     volume = strtod(argv[optind], &endptr);
-    if(*endptr != '\0') {
+    if(*endptr != '\0' && !toggle_mute) {
```
Comment (rhaas80) [inline pa_volume.c:331]: I'd rather see these lines moved into the `else` block of the `if` in line 327 and not add the `&& !toggle_mute` to this line.

**Comment by rhaas80** [comment]
Hello @amckee 

thank you for the pull request. Having control over the mute state seems like a good idea indeed.

There are some `fprintf` that should not be there for me to include it. Having used tools with a `toogle` before, I would also like to suggest to add explict `mute` and `unmute` options to make it possible to control the final state (eg to mute music when there is an incoming call and unmute afterwards irrespective of the current state). By the same token it may be good to report "[muted]" in the info output if a client is muted. 

Yours,
Roland

**Review by rhaas80 — COMMENTED** [review]
Included in git hash [eedf0ed](https://github.com/rhaas80/pa_volume/commits/eedf0ed9f10112d1a5dc6b620cd85cf241266b97) "Merge remote-tracking branch 'amckee/master' into master" of [pa_volume](https://github.com/rhaas80/pa_volume)

I added "mute" and "unmute" options as well and changed how muting is done by modifying the "mute" state (https://freedesktop.org/software/pulseaudio/doxygen/structpa__ext__stream__restore__info.html#a130e90ee7751ef01724e9b6426b76da7) of the `restore_info` struct rather than manipulating volume.

**Pull request closed without merging**